Dynatrace/dynatrace-operator
# Create new tables to improve CSI database structure

## Description

Create new tables to improve CSI database structure

DB diagram:

![image](https://github.com/Dynatrace/dynatrace-operator/assets/1374633/0918d5a2-5202-40b1-bbc6-825ef134f769)




## How can this be tested?

if you run unit test `pkg/controllers/csi/metadata/sqlite_test.go:647` TestSchemaMigration

you can validate schema migration or you can deploy it and copy sqlite db locally:
```
kubectl cp dynatrace-oneagent-csi-driver-4nmkm:/data/csi.db csi.sqlite
```

And check new tables:
```
sqlite> SELECT name FROM sqlite_master WHERE type='table';
migrations
tenant_configs
code_modules
volume_meta
os_mounts
app_mounts
```

## Checklist
